sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -1580,6 +1675,59 @@ abstract class CastBase extends UnaryExpression with TimeZoneAwareExpression wit
     }
   }
 
+  private[this] def castDayTimeIntervalToIntegralTypeCode(
+      ctx: CodegenContext,
+      endFiled: Byte,
+      integralType: String,
+      catalogType: String): CastFunction = {
+    val util = IntervalUtils.getClass.getCanonicalName.stripSuffix("$")
+    integralType match {
```

Comment:
is it just `$evPrim = $util.dayTimeIntervalTo$integralType($c, (byte)$endFiled);`?